contracts/contracts/subnet/SubnetActorManagerFacet.sol

```diff
@@ -116,14 +120,15 @@ contract SubnetActorManagerFacet is SubnetActorModifiers, ReentrancyGuard, Pausa
     ///         or equal to minimum activation collateral as a result of this operation,
     ///         then  subnet will be registered.
     /// @param publicKey The off-chain 65 byte public key that should be associated with the validator
-    function join(bytes calldata publicKey) external payable nonReentrant whenNotPaused notKilled {
+    /// @param amount The amount of collateral provided as stake
+    function join(bytes calldata publicKey, uint256 amount) external payable nonReentrant whenNotPaused notKilled {
         // Adding this check to prevent new validators from joining
         // after the subnet has been bootstrapped, if the subnet mode is not Collateral.
         // We will increase the functionality in the future to support explicit permissioning.
         if (s.bootstrapped) {
             LibSubnetActor.enforceCollateralValidation();
         }
-        if (msg.value == 0) {
+        if (amount == 0) {
```

Comment:
if the supplySource is native user is still obligated to send the msg.value number in the amount? is there a piece of logic somewhere that auto populates this field is the supplysource is native?

Reply:
yep, have to do that. This just standardises the interface. So maybe one day nft can be used as well.